src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java

```diff
@@ -18,28 +23,43 @@
         private static final String VIRTUAL_FORMAT =
             "%s" + RESOURCE_SEPARATOR + CONTEXT_SEGMENT + "%s" + RESOURCE_SEPARATOR + TENANT_TABLE_SEGMENT + "%s";
 
-        private final String context;
-        private final String tenantTableName;
+        private final String context; // must have no '/'s
+        private final String tenantTableName; // must have no '/'s
 
-        MtStreamArn(String prefix, String tableName, String streamLabel, String context, String tenantTableName) {
+        MtStreamArn(String prefix,
+            String tableName,
+            String streamLabel,
+            String slashFreeContext,
+            String slashFreeTenantTableName) {
             super(prefix, tableName, streamLabel);
-            this.context = context;
-            this.tenantTableName = tenantTableName;
+            Preconditions.checkArgument(!slashFreeContext.contains("/"),
```

Comment:
I was expecting the MtStreamArn to store the un-encoded values in memory and only encode/decode when going to/from string.